src/main/java/minijava/ir/optimize/JmpBlockRemover.java

```diff
@@ -57,10 +56,6 @@ private void remove(Block block) {
     jmpTargetNewPredecessors[jmpToTargetEdge.pos] = jmpBlockPredecessor;
     Node newJmpTarget = graph.newBlock(jmpTargetNewPredecessors);
     Graph.exchange(jmpTarget, newJmpTarget);
-
-    Node bad = graph.newBad(Mode.getANY());
-    for (int i = 0; i < block.getPredCount(); i++) {
-      block.setPred(i, bad);
-    }
+    Graph.killNode(block);
```

Comment:
Yeah, sorry for sneaking these refactorings in. But I felt it wasn't just worth the time to open a separate PR.